core/security_group.go

```diff
@@ -0,0 +1,174 @@
+package core
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/krystal/go-katapult"
+)
+
+type SecurityGroup struct {
+	ID               string   `json:"id,omitempty"`
+	Name             string   `json:"name,omitempty"`
+	AllowAllInbound  bool     `json:"allow_all_inbound,omitempty"`
+	AllowAllOutbound bool     `json:"allow_all_outbound,omitempty"`
+	Associations     []string `json:"associations,omitempty"`
+}
+
+func (sg *SecurityGroup) Ref() SecurityGroupRef {
+	return SecurityGroupRef{ID: sg.ID}
+}
+
+// allows a reference to a security group
+type SecurityGroupRef struct {
+	ID string `json:"id,omitempty"`
+}
+
+func (sg SecurityGroupRef) queryValues() *url.Values {
+	v := &url.Values{}
+	v.Set("security_group[id]", sg.ID)
+
+	return v
```

Comment:
Nitpick; As `url.Values` is just a `map[string][]string` type, and we're only ever setting one value, we can simplify this method like so:

```suggestion
	return &url.Values{
		"security_group[id]": []string{sg.ID},
	}
```

Personally I don't mind either way though, so I'll let you do with this suggestion as you wish :D